converters/fromZigbee.js

```diff
@@ -2682,7 +2682,115 @@ const converters = {
             return result;
         },
     },
+    ZNMS12LM_closuresDoorLock_change: {
+        cid: 'closuresDoorLock',
+        type: 'attReport',
+        convert: (model, msg, publish, options) => {
+            const result = {};
+            const lockStatusLookup = {
+                1: 'finger_not_match',
+                2: 'password_not_match',
+                3: 'reverse_lock', // disable open from outside
+                4: 'reverse_lock_cancel', // enable open from outside
+                5: 'locked',
+                6: 'lock_opened',
+                7: 'finger_add',
+                8: 'finger_delete',
+                9: 'password_add',
+                10: 'password_delete',
+                11: 'lock_opened_inside', // Open form inside reverse lock enbable
+                12: 'lock_opened_outside', // Open form outside reverse lock disable
+                13: 'ring_bell',
+                14: 'change_language_to',
+                15: 'finger_open',
+                16: 'password_open',
+            };
+            result.user = null;
+            result.repeat = null;
+            if (msg.data.data['65526']) { // lock final status
+                // Convert data back to hex to decode
+                const data = Buffer.from(msg.data.data['65526'], 'ascii').toString('hex');
+                const command = data.substr(6, 4);
+                if (command === '0301') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0311') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'LOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0205') {
+                    result.event = lockStatusLookup[3];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'LOCK';
+                } else if (command === '0215') {
+                    result.event = lockStatusLookup[3];
+                    result.state = 'LOCK';
+                    result.reverse = 'LOCK';
+                } else if (command === '0111') {
+                    result.event = lockStatusLookup[5];
+                    result.state = 'LOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0b00') {
+                    result.event = lockStatusLookup[12];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'UNLOCK';
```

Comment:
What does `reverse` mean?

Reply:
"reverse"  mean lock inside, and outside open disable all, i use original word from xiaomi app

Reply:
I don't understand, is it like https://www.youtube.com/watch?v=Sz-2VDfClnM?

Reply:
@Koenkk In the picture below, Yellow is like what you see in the video, Blue is the "Locked "state, Red is "Reverse locked" state, Aqara S2 is much complex. "Reverse locked" can only activated manual  inside, when you activate it, open from outside by key, finger or password all disable, you can see "Reverse locked" ativate in this https://youtu.be/bPvoQN51_5I?t=421 at 7:01
![Screenshot_20190628-201148_Mi Home](https://user-images.githubusercontent.com/10563561/60758948-80784a80-a047-11e9-936b-3ff110f10dd8.jpg)